lhotse/recipes/ami.py

```diff
@@ -0,0 +1,256 @@
+    'ES2010a', 'ES2010b', 'ES2010c', 'ES2010d', 'ES2012a', 'ES2012b', 'ES2012c', 'ES2012d', 'ES2013a', 'ES2013b',
+    'ES2013c', 'ES2013d', 'ES2014a', 'ES2014b', 'ES2014c', 'ES2014d', 'ES2015a', 'ES2015b', 'ES2015c', 'ES2015d',
+    'ES2016a', 'ES2016b', 'ES2016c', 'ES2016d', 'IB4005', 'IN1001', 'IN1002', 'IN1005', 'IN1007', 'IN1008',
+    'IN1009', 'IN1012', 'IN1013', 'IN1014', 'IN1016', 'IS1000a', 'IS1000b', 'IS1000c', 'IS1000d', 'IS1001a',
+    'IS1001b', 'IS1001c', 'IS1001d', 'IS1002b', 'IS1002c', 'IS1002d', 'IS1003a', 'IS1003b', 'IS1003c', 'IS1003d',
+    'IS1004a', 'IS1004b', 'IS1004c', 'IS1004d', 'IS1005a', 'IS1005b', 'IS1005c', 'IS1006a', 'IS1006b', 'IS1006c',
+    'IS1006d', 'IS1007a', 'IS1007b', 'IS1007c', 'IS1007d', 'TS3005a', 'TS3005b', 'TS3005c', 'TS3005d', 'TS3006a',
+    'TS3006b', 'TS3006c', 'TS3006d', 'TS3007a', 'TS3007b', 'TS3007c', 'TS3007d', 'TS3008a', 'TS3008b', 'TS3008c',
+    'TS3008d', 'TS3009a', 'TS3009b', 'TS3009c', 'TS3009d', 'TS3010a', 'TS3010b', 'TS3010c', 'TS3010d', 'TS3011a',
+    'TS3011b', 'TS3011c', 'TS3011d', 'TS3012a', 'TS3012b', 'TS3012c', 'TS3012d')
+
+split_dev = (
+    'ES2011a', 'ES2011b', 'ES2011c', 'ES2011d', 'IB4001', 'IB4002', 'IB4003', 'IB4004', 'IB4010', 'IB4011', 'IS1008a',
+    'IS1008b', 'IS1008c', 'IS1008d', 'TS3004a', 'TS3004b', 'TS3004c', 'TS3004d')
+
+split_eval = (
+    'EN2002a', 'EN2002b', 'EN2002c', 'EN2002d', 'ES2004a', 'ES2004b', 'ES2004c', 'ES2004d', 'IS1009a', 'IS1009b',
+    'IS1009c', 'IS1009d', 'TS3003a', 'TS3003b', 'TS3003c', 'TS3003d')
+
+dataset_parts = {'train': split_train, 'dev': split_dev, 'eval': split_eval}
+
+
+def download(
+        target_dir: Pathlike = '.',
+        force_download: Optional[bool] = False,
+        url: Optional[str] = 'http://groups.inf.ed.ac.uk/ami'
+) -> None:
+    assert mic == 'ihm'
+    target_dir = Path(target_dir)
+
+    # Audios
+    for part in dataset_parts:
+        for item in dataset_parts[part]:
+            headset_num = 5 if item in ('EN2001a', 'EN2001d', 'EN2001e') else 4
+            for m in range(headset_num):
+                wav_name = f'{item}.Headset-{m}.wav'
+                wav_url = f'{url}/AMICorpusMirror/amicorpus/{item}/audio/{wav_name}'
+                wav_dir = target_dir / 'wav_db' / item / 'audio'
+                wav_dir.mkdir(parents=True, exist_ok=True)
+                wav_path = wav_dir / wav_name
+                if force_download or not wav_path.is_file():
+                    urllib.request.urlretrieve(wav_url, filename=wav_path)
+
+    # Annotations
+    text_name = 'annotations.gzip'
+    text_path = target_dir / text_name
+    if force_download or not text_path.is_file():
+        text_url = f'{url}/AMICorpusAnnotations/ami_manual_annotations_v1.6.1_export.gzip'
+        urllib.request.urlretrieve(text_url, filename=text_path)
+
+
+class AmiSegmentAnnotation(NamedTuple):
+    text: str
+    begin_time: float
+    end_time: float
```

Comment:
Can you use the `Seconds` type from `lhotse.utils` (assuming it's in seconds)? Also, for consistency with other classes like `Cut`, could you name these `start` and `end`?

Reply:
Actually I think you don't need this class. You can create `lhotse.supervision.SupervisionSegment` directly, since you have all of: meeting id (can be same as recording id), channel, start, end and text. Other fields, like speaker, can be updated later if this information is not available at this stage.